dcc-portal-ui/app/scripts/compounds/js/compounds.js

```diff
@@ -360,6 +360,9 @@ angular.module('icgc.compounds.services', ['icgc.genes.models'])
           .then(function(compound) {
             _compoundEntity = _compoundEntityFactory(compound.plain());
             defer.resolve(_self);
+          }, function(){
+            Notify.hide();
+            $state.go('404', {page: 'compound', id: compoundId});
```

Comment:
@SidSyrus try this `$state.go('404', {page: 'compound', id: compoundId}, {location: false});`

docs: http://angular-ui.github.io/ui-router/site/#/api/ui.router.state.$state